src/main.rs

```diff
@@ -29,60 +75,366 @@ impl From<PrivmsgMessage> for ChatMessage {
+    fn color(&self) -> Color32 {
+        match self {
+            LogLevel::INFO => Color32::from_rgb(0, 255, 0),
+            LogLevel::WARN => Color32::from_rgb(255, 255, 0),
+            LogLevel::ERROR => Color32::from_rgb(255, 50, 0),
+        }
+    }
+}
+struct LogMessage {
+    message: String,
+    timestamp: String,
+    log_level: LogLevel,
+}
+
+struct ChatbotConfig {
     channel_name: String,
-    messages: Arc<Mutex<Vec<ChatMessage>>>,
+    auth_token: String,
+}
+
+struct Chatbot {
+    config: ChatbotConfig,
+    selected_section: Section,
+    frontend_tx: tokio::sync::mpsc::Sender<BackendMessageAction>,
+    frontend_rx: tokio::sync::mpsc::Receiver<FrontendMessageAction>,
+    labels: ChatbotUILabels,
+    log_messages: Vec<LogMessage>,
 }
 
-impl Default for Chatbot {
-    fn default() -> Self {
+impl Chatbot {
+    fn new(
+        channel_name: String,
+        auth_token: String,
+        frontend_tx: tokio::sync::mpsc::Sender<BackendMessageAction>,
+        frontend_rx: tokio::sync::mpsc::Receiver<FrontendMessageAction>,
+    ) -> Self {
         Self {
-            channel_name: String::new(),
-            messages: Arc::new(Mutex::new(Vec::new())),
+            config: ChatbotConfig {
+                channel_name: channel_name.clone(),
+                auth_token: auth_token.clone(),
+            },
+            selected_section: Section::Home,
+            frontend_tx: frontend_tx,
+            frontend_rx: frontend_rx,
+            labels: ChatbotUILabels {
+                bot_status: "Disconnected".to_string(),
+                connect_button: "Connect".to_string(),
+                tts_status: ButtonStatus::ON,
+                sfx_status: ButtonStatus::ON,
+            },
+            log_messages: Vec::new(),
         }
     }
+
+    fn show_home(&mut self, ui: &mut egui::Ui) {
+        ui.set_min_height(ui.max_rect().height());
+        ui.set_min_width(ui.max_rect().width());
+        ui.horizontal(|ui| {
+            if ui.button(&self.labels.connect_button).clicked() {
+                if self.labels.connect_button == "Connect" {
+                    if self.config.auth_token == "" {
+                        self.log_messages.push(LogMessage {
+                            message: "Tried to connect to the chat without auth token".to_string(),
+                            timestamp: chrono::Local::now().to_string(),
+                            log_level: LogLevel::ERROR,
+                        });
+                        return;
+                    }
+                    self.labels.connect_button = "Disconnect".to_string();
+                    let _ = self.frontend_tx.send(BackendMessageAction::ConnectToChat(
+                        self.config.channel_name.clone(),
+                    ));
+                    self.labels.bot_status = "Connected".to_string();
+                } else {
+                    self.labels.connect_button = "Connect".to_string();
+                    let _ = self
```

Comment:
you're not awaiting the send which is understandable in a non async context which means either going sync or blocking tokio